### DISCUSSION TITLE
data-cookiecategory on script only works the firs time?

### DISCUSSION THREAD

**Discussion by aldevit** [orestbida/cookieconsent#497] [Q&A] [answered]
data-cookiecategory on script only works the first time. If a user clicks a second time on accept or deny consent, the script is not enabled or disabled.

There is a solution?

Thanks
Al

**Comment by orestbida** [comment]
Yes, this is by design; `script` tags are executed (enabled) once.

If you want to have more control you can use the `onChange` callback, executed whenever preferences change.

**Reply by aldevit** [reply]
Thanks for the reply.

is there a command i can run  on onChange that disables or enables scripts with specific data-cookiecategory ?

**Reply by orestbida** [reply, marked as answer]
You cannot change the behavior of the `<script>` elements. There is no easy way to achieve the end result you'd like to get to, you'd need to adapt each script so that it can be used inside the [`onAccept` and `onChange` callbacks](https://github.com/orestbida/cookieconsent#available-callbacks).

E.g. the following snippet:
```html
<script type="text/plain" data-cookiecategory="analytics">
    // example code
    window.analyticsEnabled = true;
</script>
```
can be adapted to:

```javascript
const cc = initCookieConsent();

cc.run({
    // your config ...

    onAccept: () => {
        if(cc.allowedCategory('analytics')) {
            enableAnalytics()
        }
    },
    
    onChange: (cookie, changedCategories) => {
        if(changedCategories.includes('analytics')) {
            if(cc.allowedCategory('analytics')) {
                enableAnalytics();
            } else {
                // handle disable logic
                window.analyticsEnabled = false;
            }
        }
    }
});

function enableAnalytics() {
    // example code
    window.analyticsEnabled = true;
}
```

**Reply by aldevit** [reply]
Thanks orestbida
